client-java/src/unit-tests/java/uk/gov/gchq/palisade/client/FullTest.java

```diff
@@ -38,63 +40,109 @@
 @MicronautTest
 class FullTest {
```

Comment:
This could be debated I suppose. Maybe a contract test and should be moved out to be more in line with our palisade services tests.

Reply:
Contract deals with expected external comms made by the code.
Component deals with multiple-classes 'big-picture' logic of the code, minus any external comms.
Unit deals with one method at a time.

If you are using a mocked palisade, frs and data service, the test is probably contract.
Otherwise, you have a pretty clear-cut unit-test or not unit-test => component

Reply:
That test has now been moved to it's new home in `src/contract-tests/java` as it exercises the API from the outside.